asciidoc/modules/con_mg_migrating-clusters.adoc

```diff
@@ -0,0 +1,40 @@
+[id="migrating_clusters_{context}"]
+
+= Migrating clusters
+
+It is important to decide the migration strategy for your codebase in a cluster. This is because you cannot add {VertX} {v3x} nodes and {VertX} {v4} nodes together in a single cluster for the following reasons: 
+
+* Cluster manager upgrades - If there are major version upgrades in cluster managers, they may not have support all features for backward compatibility. You will have to check the updates in cluster managers.
+* Subscription data changes - If the information stored in subscription maps or caches is no longer the same in the cluster manager versions. 
+* Transport protocol changes - If some fields in the message transport protocol have changed in the cluster.
+
+If you have a {VertX} cluster for a single application or for some closely related microservices, you can migrate the entire codebase to the new cluster at one time.
+
+However, if you cannot migrate the codebase at one time, use the recommendations in this section to migrate a {VertX} {v3x} codebase to {VertX} {v4}.
+
+== Splitting the cluster
+
+If you have a cluster where different teams have deployed verticles for their applications, you can consider splitting the {VertX} {v3x} cluster into smaller ones. Note that after splitting the cluster, the separated components will not be able to communicate using the clustering features. You can splitthe cluster using the following XXX:
+
+* EventBus request and reply - HTTP/RESTful web services, gRPC
+* EventBus send and publish - Messaging systems such as, Kafka, Postgres `LISTEN` and `NOTIFY`, Redis Pub and Sub
+* Shared Data - Redis, Infinispan
```

Comment:
```suggestion
* Shared Data - Redis, Infinispan, Hazelcast
```

Reply:
As discussed, adding an attribute to show "Hazelcast" only in community documentation.